File: Insert @done.lua
--[[
VPLanguage = lua
VPScriptSuperMenuTitle = GTD
VPScriptMenuTitle = Insert @done
VPShortcutKey = d
VPShortcutMask = control
VPEndConfig
--]]

-- check out http://www.lua.org/pil/22.1.html for more formatting options.

textView = windowController:textView()
document = windowController:document()

if textView ~= nil then
    textView:insertText("@done:")
end

File: Insert @done.py
# -*- coding: utf-8 -*-

'''
:Title
Insert @done

:Planguage
Python

:Requires
VoodooPad 3.5+

:Description
Inserts "@done:" in front of the given text.

EOD
'''
VPScriptSuperMenuTitle = "GTD"
VPScriptMenuTitle = "Insert @done"
VPShortcutMask = "control"
VPShortcutKey = "d"

import AppKit

def main(windowController, *args, **kwargs):
    textView = windowController.textView()
    document = windowController.document()

    if textView != None:
        textView.insertText_("@done:")
